File: src/types/model/card.ts
export interface Card {
  id: number;
  user_id: number;
  card_number: string;
  card_type: string;
  expire_date: Date;
  cvv: string;
  card_provider: string;
  created_at: string;
  updated_at: string;
}

export interface CardTrashed extends Card {
  deleted_at: string;
}

export interface DashboardCard {
  total_balance?: number | null;
  total_topup?: number | null;
  total_withdraw?: number | null;
  total_transaction?: number | null;
  total_transfer?: number | null;
}

export interface DashboardCardCardNumber {
  total_balance?: number | null;
  total_topup?: number | null;
  total_withdraw?: number | null;
  total_transaction?: number | null;
  total_transfer_send?: number | null;
  total_transfer_receiver?: number | null;
}

export interface CardMonthBalance {
  month: string;
  total_balance: number;
}

export interface CardYearlyBalance {
  year: string;
  total_balance: number;
}

export interface CardMonthTopupAmount {
  month: string;
  total_amount: number;
}

export interface CardYearlyTopupAmount {
  year: string;
  total_amount: number;
}

export interface CardMonthTransactionAmount {
  month: string;
  total_amount: number;
}

export interface CardYearlyTransactionAmount {
  year: string;
  total_amount: number;
}

export interface CardMonthTransferAmount {
  month: string;
  total_amount: number;
}

export interface CardYearlyTransferAmount {
  year: string;
  total_amount: number;
}

export interface CardMonthWithdrawAmount {
  month: string;
  total_amount: number;
}

export interface CardYearlyWithdrawAmount {
  year: string;
  total_amount: number;
}

File: src-tauri/src/model/card.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct CardResponse {
    pub id: i32,
    pub user_id: i32,
    pub card_number: String,
    pub card_type: String,
    pub expire_date: String,
    pub cvv: String,
    pub card_provider: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct CardResponseDeleteAt {
    pub id: i32,
    pub user_id: i32,
    pub card_number: String,
    pub card_type: String,
    pub expire_date: String,
    pub cvv: String,
    pub card_provider: String,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct DashboardCard {
    pub total_balance: Option<i64>,
    pub total_topup: Option<i64>,
    pub total_withdraw: Option<i64>,
    pub total_transaction: Option<i64>,
    pub total_transfer: Option<i64>,
}

#[derive(Serialize, Deserialize)]
pub struct DashboardCardCardNumber {
    pub total_balance: Option<i64>,
    pub total_topup: Option<i64>,
    pub total_withdraw: Option<i64>,
    pub total_transaction: Option<i64>,
    pub total_transfer_send: Option<i64>,
    pub total_transfer_receiver: Option<i64>,
}

#[derive(Serialize, Deserialize)]
pub struct CardResponseMonthBalance {
    pub month: String,
    pub total_balance: i64,
}

#[derive(Serialize, Deserialize)]
pub struct CardResponseYearlyBalance {
    pub year: String,
    pub total_balance: i64,
}

#[derive(Serialize, Deserialize)]
pub struct CardResponseMonthAmount {
    pub month: String,
    pub total_amount: i64,
}

#[derive(Serialize, Deserialize)]
pub struct CardResponseYearAmount {
    pub year: String,
    pub total_amount: i64,
}

